File: Be-siukat-new/siukat-service-dev_snmptn_2022/models/ukt.js
var Sequelize = require('sequelize')
var db = require('../config/database')

const Ukt = db.define('ukt', {
  id_ukt:{
  	type: Sequelize.INTEGER,
  	primaryKey: true
  },
  major_id: {
    type: Sequelize.STRING
  },
  entrance: {
    type: Sequelize.INTEGER
  },
  semester: {
    type: Sequelize.INTEGER
  },
  status: {
    type: Sequelize.ENUM('aktif', 'nonaktif')
  },
  degree: {
    type: Sequelize.STRING
  },
  decree: {
    type: Sequelize.STRING
  },
  I: {
    type: Sequelize.INTEGER
  },
  II: {
    type: Sequelize.INTEGER
  },
  III: {
    type: Sequelize.INTEGER
  },
  IV: {
    type: Sequelize.INTEGER
  },
  V: {
    type: Sequelize.INTEGER
  },
  VI: {
    type: Sequelize.INTEGER
  },
  VII: {
    type: Sequelize.INTEGER
  },
  VIII: {
    type: Sequelize.INTEGER
  }
},
 {
  // don't add the timestamp attributes (updatedAt, createdAt)
  timestamps: false,

  // don't delete database entries but set the newly added attribute deletedAt
  // to the current date (when deletion was done). paranoid will only work if
  // timestamps are enabled
  paranoid: true,

  // don't use camelcase for automatically added attributes but underscore style
  // so updatedAt will be updated_at
  underscored: true,

  // disable the modification of tablenames; By default, sequelize will automatically
  // transform all passed model names (first parameter of define) into plural.
  // if you don't want that, set the following
  freezeTableName: true,

  // define the table's name
  tableName: 'ref_ukt'
}
);
module.exports = Ukt

File: BackEnd-Siukat/models/ukt.go
package models

type Ukt struct {
	IDUkt         int    `gorm:"primaryKey;column:id_ukt" json:"id_ukt"`
	MajorID       string `gorm:"column:major_id;type:varchar(255)" json:"major_id"`
	Entrance      int    `gorm:"column:entrance" json:"entrance"`
	Semester      int    `gorm:"column:semester" json:"semester"`
	Status        string `gorm:"column:status;type:enum('aktif','nonaktif')" json:"status"`
	Degree        string `gorm:"column:degree;type:varchar(255)" json:"degree"`
	Decree        string `gorm:"column:decree;type:varchar(255)" json:"decree"`
	Bkt           int    `gorm:"column:bkt" json:"bkt"`
	I             int    `gorm:"column:I" json:"I"`
	II            int    `gorm:"column:II" json:"II"`
	III           int    `gorm:"column:III" json:"III"`
	IV            int    `gorm:"column:IV" json:"IV"`
	V             int    `gorm:"column:V" json:"V"`
	VI            int    `gorm:"column:VI" json:"VI"`
	VII           int    `gorm:"column:VII" json:"VII"`
	VIII          int    `gorm:"column:VIII" json:"VIII"`
	Bm            int    `gorm:"column:bm" json:"bm"`
	Kjmu          int    `gorm:"column:kjmu" json:"kjmu"`
	Internasional int    `gorm:"column:internasional" json:"internasional"`
	Nominal       int    `gorm:"-" json:"nominal"` // Virtual field for frontend
}

func (Ukt) TableName() string {
	return "ref_ukt"
}
